{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def missing_gt(samples, processed):\n",
    "    return [i for i in samples if not i in processed]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def harvest_gt(gt_folder, maxchr=30):\n",
    "    F1 = os.listdir(gt_folder)\n",
    "    samples = [i.split(\".\")[0] for i in F1 if i.endswith(\"genotype\")]\n",
    "    folders1 = [i for i in F1 if i.endswith(\"DIR\")]\n",
    "    nothing = [i for i in samples if not i+\".genotype_DIR\" in folders1]\n",
    "    \n",
    "    sorted_files = []\n",
    "    split_files = []\n",
    "    flipped_files = []\n",
    "    \n",
    "    for folder in folders1:\n",
    "        lf = os.path.join(gt_folder, folder)\n",
    "        ldir = os.listdir(lf)\n",
    "        sortedgt = [i for i in ldir if i.endswith(\".sorted\")]\n",
    "        chr_1s = [i for i in ldir if i.endswith(\".chr.{}\".format(maxchr))]\n",
    "        if len(chr_1s)>=1:\n",
    "            split_files.append(chr_1s[0].split(\".\")[0])\n",
    "            if len(chr_1s)>1:\n",
    "                flipped_files.append(chr_1s[0].split(\".\")[0])\n",
    "        if len(sortedgt)>=1:\n",
    "            sorted_files.append(sortedgt[0].split(\".\")[0]) \n",
    "    \n",
    "    missing_sort = missing_gt(samples, sorted_files)\n",
    "    missing_split = missing_gt(samples, split_files)\n",
    "    missing_flip = missing_gt(samples, flipped_files)\n",
    "\n",
    "    jobstate_dict = {\"not_sorted\":missing_sort,\n",
    "                     \"not_split\":missing_split,\n",
    "                     \"not_flipped\":missing_flip}\n",
    "    return  nothing, samples,jobstate_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def harvest_not_completely_done(samples, chromosomes, windowsize, gtdir, tgdir):\n",
    "    F2 = [i.split(\".\")[0] for i in os.listdir(tgdir)]\n",
    "    \n",
    "    # create dict for the ones that dont have any data at all\n",
    "    completely_no_tig = missing_gt(samples, F2)\n",
    "    fresh_not_done_all = {}\n",
    "    for folder in completely_no_tig:\n",
    "        fresh_not_done_all.setdefault(folder,{})\n",
    "        #for windowsize in windowsizes:\n",
    "        fresh_not_done_all[folder].setdefault(str(windowsize), [])  #add indent here for retooling\n",
    "        for chromosome in chromosomes: #add indent here for retooling\n",
    "            fresh_not_done_all[folder][str(windowsize)].append(str(chromosome)) #add indent here for retooling\n",
    "    \n",
    "    # create dictionary with all files that are supposed to be there.\n",
    "    supp_all = {}\n",
    "    for folder in F2:\n",
    "        supp_all.setdefault(folder,{})\n",
    "        #for windowsize in windowsizes:\n",
    "        supp_all[folder].setdefault(str(windowsize), []) #add indent here for retooling\n",
    "        for chromosome in chromosomes:  #add indent here for retooling\n",
    "\n",
    "            #supp_sample = \"{}.genotype.{}.rough_COs_windowsize{}.refined.recomb.txt\".format(folder, chromosome, windowsize)\n",
    "            #supp_all[folder][str(windowsize)].append(supp_sample)\n",
    "            supp_all[folder][str(windowsize)].append(str(chromosome))  #add indent here for retooling\n",
    "\n",
    "    # create dictionary with all files that are there\n",
    "    present_all = {}\n",
    "    for folder in F2:\n",
    "        #print(folder)\n",
    "        fpF2 = os.path.join(tgdir,folder+\".genotype\")\n",
    "        #print(fpF2)\n",
    "        actually_there = [i for i in os.listdir(fpF2) if i.endswith(\"refined.recomb.txt\")]\n",
    "        #print(actually_there)\n",
    "        for file in actually_there:\n",
    "            split_file = file.split(\".\")\n",
    "            if \"flipped\" in split_file:\n",
    "                pass  #to be implemented later\n",
    "            else:\n",
    "                sample = split_file[0]\n",
    "                chromosome = split_file[2] \n",
    "                windowsize = split_file[3].lstrip(\"rough_COs_windowsize\")\n",
    "                present_all.setdefault(sample,{}).setdefault(str(windowsize),[]).append(chromosome)\n",
    "                #present_all.setdefault(sample,{}).setdefault(windowsize,[]).append(file)\n",
    "    \n",
    "    # create dictionary with files that are missing\n",
    "    needed_to_be_done = {}\n",
    "    for sample, item in supp_all.items():\n",
    "        if not sample in present_all.keys():\n",
    "            #print(\"sample is not present at all, redo {}\".format(sample))\n",
    "            needed_to_be_done[sample]=item\n",
    "        else:\n",
    "            #print(\"processing sample {}\".format(sample))\n",
    "            for windowsize, item2 in item.items():\n",
    "                #print(\"    processing windowsize {}\".format(windowsize))\n",
    "                if not windowsize in present_all[sample].keys():\n",
    "                    #print(\"        windowsize {} not present, adding whole\".format(windowsize))\n",
    "                    needed_to_be_done.setdefault(sample, {})[windowsize]=item2\n",
    "                else:\n",
    "                    for item3 in item2:\n",
    "                        if not item3 in dict.fromkeys(present_all[sample][windowsize]):\n",
    "                            needed_to_be_done.setdefault(sample, {}).setdefault(windowsize, []).append(item3)\n",
    "    \n",
    "    return fresh_not_done_all, needed_to_be_done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sort_job(sample_name, target_dir):\n",
    "    inputfile  = os.path.join(target_dir,\n",
    "                              sample_name+\".genotype\")\n",
    "    outputfile = os.path.join(target_dir,\n",
    "                              sample_name+\".genotype_DIR\",\n",
    "                              sample_name+\".genotype.sorted\")\n",
    "    \n",
    "    jobstring  = \"sort -k 1,1n -k 2,2n {inputfile} > {outputfile}\".format(inputfile=inputfile, outputfile=outputfile)\n",
    "    return jobstring\n",
    "\n",
    "def create_split_job(sample_name, target_dir, index, maxchr=30):\n",
    "    f = sample_name+\".genotype\"\n",
    "    input_gt_file = os.path.join(target_dir,\n",
    "                                 f+\"_DIR\",\n",
    "                                 f+\".sorted\")\n",
    "    outpath = os.path.join(target_dir,f+\"_DIR\")\n",
    "    jobstring = \"python scripts/vcf2individualGenotypes/split_by_chrom.py -i {input_gt_file}  -o {outpath}  -s {index}  -m {maxchr} -d {f}\".format(input_gt_file=input_gt_file, outpath=outpath,index=index,maxchr=maxchr,f=f)\n",
    "    return jobstring\n",
    "\n",
    "def create_flip_job(sample_name, target_dir):\n",
    "    pass\n",
    "\n",
    "\n",
    "def create_caller(chromosome, sample_name, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    cmdCaller = \"scripts/individualGenotypes2breaks/base_caller.jar\"\n",
    "    in_1 = os.path.join(gt_dir, f+\"_DIR\", f+\".chr.{chr}\".format(chr=str(chromosome)))\n",
    "    caller_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_allele_co.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring = \"java -Xmx4G -jar  {cmdCaller}  -r {in_1} -o {caller_out} -n bi \".format(cmdCaller=cmdCaller,\n",
    "                                                                                           in_1=in_1,\n",
    "                                                                                           caller_out=caller_out)\n",
    "    return jobstring\n",
    "    \n",
    "def create_estimate(chromosome, sample_name, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    in_1  = os.path.join(gt_dir, f+\"_DIR\", f+\".chr.{chr}\".format(chr=chromosome))\n",
    "    cmdEstimate  = \"scripts/individualGenotypes2breaks/allele_freq_estimator.jar\"\n",
    "    out_estimate = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_frequencies_for_hmm.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring =  \"java -Xmx4G -jar {cmdEstimate}  -r {in_1} -o {out_estimate} -n bi -w {winsize} \".format(cmdEstimate=cmdEstimate,\n",
    "                                                                                                               in_1=in_1,\n",
    "                                                                                                               winsize=winsize,\n",
    "                                                                                                               out_estimate=out_estimate)\n",
    "    return jobstring\n",
    "\n",
    "def create_beta(chromosome, sample_name, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    in_1  = os.path.join(gt_dir, f+\"_DIR\", f+\".chr.{chr}\".format(chr=chromosome))\n",
    "    cmdBeta = \"scripts/individualGenotypes2breaks/beta_mixture_model.R\"\n",
    "    # beta_in is estimate_out\n",
    "    beta_in = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_frequencies_for_hmm.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    beta_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_bmm.intersections.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring = \"Rscript --vanilla {cmdBeta} {beta_in} {beta_out} \".format(cmdBeta=cmdBeta, beta_in=beta_in, beta_out=beta_out)\n",
    "    return jobstring\n",
    "\n",
    "def create_prep_prob(chromosome, sample_name,genomesize, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    in_1  = os.path.join(gt_dir, f+\"_DIR\", f+\".chr.{chr}\".format(chr=chromosome))\n",
    "    cmdPrep = \"scripts/individualGenotypes2breaks/prep_prob.pl\"\n",
    "    caller_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_allele_co.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    prep_prob_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_file_for_probabilities.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring = \"perl {cmdPrep} -s {sample_name} -m {in_1} -b {caller_out} -c {genomesize} -o {prep_prob_out} \".format(cmdPrep=cmdPrep, sample_name=sample_name, in_1=in_1, caller_out=caller_out, genomesize=genomesize, prep_prob_out=prep_prob_out)\n",
    "\n",
    "    return jobstring\n",
    "\n",
    "def create_hmm_prob(chromosome, sample_name,genomesize, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    cmdHmm = \"scripts/individualGenotypes2breaks/hmm_prob.pl\"\n",
    "    outshort =  os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    estimate_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_frequencies_for_hmm.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    prep_prob_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_file_for_probabilities.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    beta_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_bmm.intersections.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring = \"perl {cmdHmm}  -s {estimate_out} -p {prep_prob_out} -o {outshort} -a {beta_out} -c {genomesize} \".format(cmdHmm=cmdHmm, estimate_out=estimate_out, prep_prob_out=prep_prob_out, outshort=outshort, beta_out=beta_out, genomesize=genomesize)\n",
    "    return jobstring\n",
    "\n",
    "\n",
    "# os.path.join(config[\"TIGER_WD\"], \"{study}\" + config[\"TIGER_SUFFIX\"], \"{file}\", \"{file2}.{chr}.windowsize{winsize,[0-9]+}hmm_model\")\n",
    "def create_hmm(chromosome, sample_name, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    cmdHmm = \"scripts/individualGenotypes2breaks/hmm_play.jar\"\n",
    "    caller_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_allele_co.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    hmm_prep_out = os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}hmm_model\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    hmm_out =os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_hmm.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring = \"java -jar {cmdHmm} -r {caller_out} -o {hmm_out} -t bi -z {hmm_prep_out} \".format(cmdHmm=cmdHmm,caller_out=caller_out,hmm_out=hmm_out,hmm_prep_out=hmm_prep_out)\n",
    "\n",
    "    return jobstring\n",
    "\n",
    "def create_prep_co(chromosome, sample_name,genomesize, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    in_1 = os.path.join(gt_dir, f+\"_DIR\", f+\".chr.{chr}\".format(chr=chromosome))\n",
    "    cmdPrepCo =\"scripts/individualGenotypes2breaks/prepare_break.pl\"\n",
    "    hmm_out =os.path.join(target_dir, f,f+\".{chromosome}.windowsize{winsize}_hmm.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    prep_co_total = os.path.join(target_dir, f,f+\".{chromosome}.rough_COs_windowsize{winsize}.txt\".format(chromosome=chromosome, winsize=winsize))\n",
    "    jobstring = \"perl {cmdPrepCo} -s {sample_name} -m {in_1} -b {hmm_out} -c  {genomesize} -o {prep_co_total} \".format(cmdPrepCo=cmdPrepCo,prep_co_total=prep_co_total, sample_name=sample_name,in_1=in_1, genomesize=genomesize, hmm_out=hmm_out)\n",
    "\n",
    "    return jobstring\n",
    "\n",
    "def create_smooth_prep_co(chromosome, sample_name, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    cmdSmooth = \"scripts/individualGenotypes2breaks/breaks_smoother.pl\"\n",
    "    prep_partial = os.path.join(target_dir, f,f+\".{chromosome}.rough_COs_windowsize{winsize}.breaks.txt\".format(chromosome=chromosome, winsize=winsize))\n",
    "    smooth_out = os.path.join(target_dir, f,f+\".{chromosome}.corrected.refined.breaks_windowsize{winsize}.txt\".format(chromosome=chromosome, winsize=winsize) )\n",
    "    jobstring = \"perl {cmdSmooth} -b {prep_partial} -o {smooth_out} \".format(cmdSmooth=cmdSmooth, prep_partial=prep_partial, smooth_out=smooth_out)\n",
    "    return jobstring\n",
    "\n",
    "def create_refine(chromosome, sample_name, target_dir, gt_dir,winsize=200):\n",
    "    f = sample_name+\".genotype\"\n",
    "    in_1 = os.path.join(gt_dir, f+\"_DIR\", f+\".chr.{chr}\".format(chr=chromosome))\n",
    "    cmdRefine = \"scripts/individualGenotypes2breaks/refine_recombination_break.pl\"\n",
    "    prep_partial = os.path.join(target_dir, f,f+\".{chromosome}.rough_COs_windowsize{winsize}.breaks.txt\".format(chromosome=chromosome, winsize=winsize))\n",
    "    jobstring = \"perl {cmdRefine} {in_1} {prep_partial}\".format(cmdRefine=cmdRefine, in_1=in_1, prep_partial=prep_partial)\n",
    "    return jobstring\n",
    "\n",
    "\n",
    "def create_tiger_job(chromosome, sample_name, target_dir, genomesize, gt_dir,winsize=200):\n",
    "    jobstrings = []\n",
    "    jobstrings.append(create_caller(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize))\n",
    "    jobstrings.append(create_estimate(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize))\n",
    "    jobstrings.append(create_beta(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize))\n",
    "    jobstrings.append(create_prep_prob(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize, genomesize=genomesize))\n",
    "    jobstrings.append(create_hmm_prob(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize, genomesize=genomesize))\n",
    "    jobstrings.append(create_hmm(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize))\n",
    "    jobstrings.append(create_prep_co(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize, genomesize=genomesize))\n",
    "    jobstrings.append(create_smooth_prep_co(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize))\n",
    "    jobstrings.append(create_refine(chromosome=chromosome, sample_name=sample_name, target_dir=target_dir, gt_dir=gt_dir,winsize=winsize))\n",
    "    return jobstrings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sbatch(joblines,name,conda_env=\"v3\", hours=\"8\",minutes=\"00\", seconds=\"00\", cores=\"2\", user =\"tilman.ronneburg@imbim.uu.se\", proj=\"snic2017-7-53\" ):\n",
    "    sbatch_header = [ \"#!/bin/bash -l\",\n",
    "                        \"#SBATCH -A {proj}\".format(proj=proj),\n",
    "                        \"#SBATCH -t {hours}:{minutes}:{seconds} \".format(hours=hours, minutes=minutes, seconds=seconds),\n",
    "                        \"#SBATCH -p core -n {cores}\".format(cores=cores),\n",
    "                        \"#SBATCH -J {name}\".format(name=name.rsplit(\".\",1)[0].rsplit(\"/\",1)[1]),\n",
    "                        \"#SBATCH -o {name}_%j.out\".format(name=name.rsplit(\".\",1)[0].rsplit(\"/\",1)[1]),\n",
    "                        \"#SBATCH -e {name}_%j.error\".format(name=name.rsplit(\".\",1)[0].rsplit(\"/\",1)[1]),\n",
    "                        \"#SBATCH --mail-user {user}\".format(user=user),\n",
    "                        \"#SBATCH --mail-type=FAIL\",\n",
    "                        \"#SBATCH --get-user-env\",\n",
    "                        \"##command underneath this##\"\n",
    "                       ]\n",
    "    load_modules = [\"source activate {env}\".format(env=conda_env)]\n",
    "    \n",
    "    with open(name, \"w\") as handle:\n",
    "        for line in sbatch_header:\n",
    "            handle.write(line+\"\\n\")\n",
    "        for line in load_modules:\n",
    "            handle.write(line+\"\\n\")\n",
    "        for line in joblines:\n",
    "            handle.write(line+\"\\n\")\n",
    "    subprocess.call([\"chmod\", \"766\", str(name)])\n",
    "    \n",
    "def create_batchfiles(jobdict, scriptdir, prefix=\"batch_\", execute=False):\n",
    "    for sample, jobs in jobdict.items():\n",
    "        print(sample)\n",
    "        print(len(jobs))\n",
    "        name = os.path.join(scriptdir, \"{prefix}{sample}.job\".format(prefix=prefix, sample=sample))\n",
    "        #print(name)\n",
    "        create_sbatch(joblines=jobs, name=name) # remember that this function has a lot of default options\n",
    "        if execute==True:\n",
    "            subprocess.call([\"sbatch\", name])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "tgdir = \"/home/tilman/storage3/process_stripes/stripes_optimise_F7/with.fam.f2.call2.TIGER_OUT/\"\n",
    "gtdir = \"/home/tilman/storage3/process_stripes/stripes_optimise_F7/with.fam.f2.call2.Genotype/\"\n",
    "gsize =  \"/home/tilman/storage3/stripes_optimise/aux/size.genome\"\n",
    "index =  \"/home/tilman/storage3/stripes_optimise/aux/Index_fastq_contig_chr.txt\"\n",
    "winsize = 50\n",
    "chromosomes = list(range(1,31))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "nothing, samples, jstate = harvest_gt(gt_folder=gtdir, maxchr=30)\n",
    "a, b = harvest_not_completely_done(samples=samples, chromosomes=chromosomes, windowsize=winsize, gtdir=gtdir, tgdir=tgdir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#jstate\n",
    "#a\n",
    "#b"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "jobdict = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "#jstate is empty, so we can ignore this thing\n",
    "for sample in jstate[\"not_sorted\"]:\n",
    "    jobdict[sample] = []\n",
    "    jobdict[sample].append(\"mkdir {}\".format(os.path.join(gtdir,sample+\".genotype_DIR\")))\n",
    "    jobdict[sample].append(create_sort_job(sample_name=sample, target_dir=gtdir))\n",
    "    jobdict[sample].append(create_split_job(sample_name=sample, target_dir=gtdir, index=index))\n",
    "    jobdict[sample].append(\"mkdir {}\".format(os.path.join(tgdir,sample+\".genotype\")))\n",
    "    for chromosome in chromosomes:\n",
    "        tiger_string = \" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize))\n",
    "        jobdict[sample].append(tiger_string)\n",
    "        \n",
    "for sample in jstate[\"not_split\"]:\n",
    "    if not sample in jobdict.keys():\n",
    "        jobdict[sample] = []\n",
    "        jobdict[sample].append(create_split_job(sample_name=sample, target_dir=gtdir, index=index))\n",
    "        jobdict[sample].append(\"mkdir {}\".format(os.path.join(tgdir,sample+\".genotype\")))\n",
    "        for chromosome in chromosomes:\n",
    "            tiger_string = \" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize))\n",
    "            jobdict[sample].append(tiger_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "#a is empty so we can ignore this\n",
    "for sample in a.keys():\n",
    "    if not sample in jobdict.keys():\n",
    "        jobdict[sample] = []\n",
    "        jobdict[sample].append(\"mkdir {}\".format(os.path.join(tgdir,sample+\".genotype\")))\n",
    "        for chromosome in chromosomes:\n",
    "            tiger_string = \" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize))\n",
    "            jobdict[sample].append(tiger_string)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "2\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n",
      "1\n"
     ]
    }
   ],
   "source": [
    "jobdict = {}\n",
    "dict24 = []\n",
    "shortjobs = {}\n",
    "for sample, item in b.items():\n",
    "    if not sample in jobdict.keys():\n",
    "        #jobdict[sample] = []\n",
    "        for windowsize, item2 in item.items():\n",
    "            if len(item2)>8:\n",
    "                jobdict[sample] = []\n",
    "                for chromosome in item2:\n",
    "                    if not chromosome == \"24\":\n",
    "                        jobdict[sample].append(\" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize)))\n",
    "                    else :\n",
    "                        dict24.append(\" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize)))\n",
    "            else:\n",
    "                print(len(item2))\n",
    "                shortjobs[sample] = []\n",
    "                for chromosome in item2:\n",
    "                    if not chromosome == \"24\":\n",
    "                        shortjobs[sample].append(\" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize)))\n",
    "                    else :\n",
    "                        dict24.append(\" && \".join(create_tiger_job(chromosome=chromosome,sample_name=sample, target_dir=tgdir,genomesize=gsize, gt_dir=gtdir,winsize=winsize)))\n",
    "                "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for k,i in jobdict.items():\n",
    "    #print(len(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "#dict24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "scriptdir = '/home/tilman/storage3/process_stripes/stripes_optimise_F7/Stripes/stripes'\n",
    "name = os.path.join(scriptdir, \"{prefix}{sample}.job\".format(prefix='all_chr_24_200', sample=\"mixed_samples\"))\n",
    "create_sbatch(joblines=dict24, name=name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "290315 0\n",
      "297115 0\n",
      "298215 0\n",
      "238115 0\n",
      "321615 0\n",
      "302815 0\n",
      "327215 0\n",
      "286815 0\n",
      "310915 0\n",
      "311115 0\n",
      "275215 0\n",
      "280015 0\n",
      "305915 0\n",
      "288115 0\n",
      "246715 0\n",
      "333515 0\n",
      "278915 0\n",
      "308215 0\n",
      "304015 0\n",
      "325415 0\n",
      "301515 0\n",
      "302315 0\n",
      "313315 0\n",
      "305515 0\n",
      "253815 0\n",
      "271515 0\n",
      "238415 0\n",
      "260215 0\n",
      "309215 0\n",
      "251015 0\n",
      "255215 1\n",
      "273915 0\n",
      "258015 0\n",
      "256415 0\n",
      "244815 0\n",
      "255315 0\n",
      "291615 0\n",
      "248915 0\n",
      "318715 0\n",
      "325515 0\n",
      "235315 0\n",
      "257015 0\n",
      "296015 0\n",
      "295315 0\n",
      "235015 0\n",
      "283515 0\n",
      "303715 0\n",
      "315615 0\n",
      "237815 0\n",
      "283615 0\n",
      "236315 0\n",
      "260615 0\n",
      "243115 4\n",
      "307115 0\n",
      "302415 0\n",
      "320015 0\n",
      "322315 0\n",
      "271715 0\n",
      "285515 0\n",
      "291715 0\n",
      "318215 0\n",
      "294415 0\n",
      "330415 0\n",
      "253915 0\n",
      "246815 0\n",
      "331415 1\n",
      "250315 0\n",
      "329515 0\n",
      "237615 0\n",
      "301415 0\n",
      "234215 1\n",
      "325115 0\n",
      "287315 0\n",
      "240415 2\n",
      "314815 0\n",
      "280515 0\n",
      "278115 0\n",
      "311515 0\n",
      "257815 0\n",
      "296415 0\n",
      "246515 0\n",
      "329615 0\n",
      "312515 0\n",
      "295515 0\n",
      "276815 0\n",
      "273415 0\n",
      "302615 0\n",
      "293615 0\n",
      "261915 0\n",
      "302915 0\n",
      "301115 0\n",
      "249215 0\n",
      "310715 0\n",
      "312615 0\n",
      "247415 0\n",
      "319215 0\n",
      "296215 0\n",
      "241915 0\n",
      "292815 0\n",
      "298115 0\n",
      "313615 1\n",
      "306615 0\n",
      "276215 0\n",
      "308815 1\n",
      "273715 0\n",
      "294915 0\n",
      "252415 0\n",
      "286615 4\n",
      "282715 0\n",
      "311715 0\n",
      "287915 0\n",
      "242415 1\n",
      "319015 0\n",
      "277915 0\n",
      "251515 0\n",
      "329015 0\n",
      "331315 0\n",
      "256515 0\n",
      "281015 0\n",
      "305715 0\n",
      "272415 0\n",
      "287115 0\n",
      "305215 0\n",
      "272015 0\n",
      "312715 0\n",
      "237215 0\n",
      "293015 0\n",
      "317515 0\n",
      "238015 0\n",
      "242715 4\n",
      "316815 0\n",
      "332715 0\n",
      "317015 0\n",
      "291515 2\n",
      "245115 0\n",
      "281815 0\n",
      "317815 0\n",
      "295915 0\n",
      "241115 0\n",
      "331915 0\n",
      "245715 0\n",
      "251215 1\n",
      "317615 0\n",
      "299715 0\n",
      "289015 0\n",
      "242615 0\n",
      "304615 0\n",
      "304715 0\n",
      "255615 0\n",
      "320515 0\n",
      "239015 0\n",
      "291215 1\n",
      "280715 0\n",
      "255515 0\n",
      "275315 0\n",
      "235115 0\n",
      "248615 1\n",
      "323415 0\n",
      "309915 0\n",
      "260815 0\n",
      "326915 0\n",
      "296915 0\n",
      "334415 0\n",
      "242215 0\n",
      "286115 3\n",
      "286315 0\n",
      "283915 0\n",
      "309615 0\n",
      "247315 0\n",
      "322715 0\n",
      "312215 0\n",
      "235915 0\n",
      "332415 0\n",
      "326715 0\n",
      "330615 0\n",
      "233015 0\n",
      "300815 0\n",
      "329715 0\n",
      "332915 0\n",
      "261315 0\n",
      "243515 0\n",
      "316315 0\n",
      "234915 0\n",
      "292215 0\n",
      "254915 0\n",
      "280415 0\n",
      "331015 0\n",
      "330115 0\n",
      "303215 0\n",
      "303515 0\n",
      "294215 0\n",
      "300215 0\n",
      "322415 0\n",
      "279015 0\n",
      "239415 3\n",
      "285015 0\n",
      "243015 3\n",
      "306715 0\n",
      "259715 0\n",
      "235715 0\n",
      "248215 3\n",
      "304515 0\n",
      "291815 0\n",
      "272315 7\n",
      "294115 1\n",
      "289915 0\n",
      "276915 0\n",
      "312415 0\n",
      "317315 0\n",
      "329815 0\n",
      "305315 0\n",
      "313915 0\n",
      "293415 0\n",
      "310515 0\n",
      "292115 0\n",
      "287715 0\n",
      "248315 0\n",
      "285215 0\n",
      "278815 0\n",
      "325815 0\n",
      "327015 0\n",
      "279515 0\n",
      "277815 0\n",
      "240315 0\n",
      "307515 0\n",
      "314915 0\n",
      "235515 0\n",
      "332315 0\n",
      "283815 0\n",
      "319115 0\n",
      "254815 0\n",
      "334115 0\n",
      "288415 0\n",
      "302015 0\n",
      "240115 0\n",
      "272715 0\n",
      "251615 0\n",
      "310015 0\n",
      "314615 0\n",
      "323015 0\n",
      "240815 0\n",
      "238315 0\n",
      "234615 0\n",
      "233415 0\n",
      "274815 0\n",
      "280815 0\n",
      "255715 0\n",
      "321115 0\n",
      "306515 0\n",
      "251415 0\n",
      "283715 0\n",
      "308915 0\n",
      "276115 0\n",
      "319515 0\n",
      "326015 0\n",
      "255915 0\n",
      "333415 0\n",
      "311815 0\n",
      "249015 0\n",
      "237515 0\n",
      "248415 0\n",
      "276615 0\n",
      "307015 0\n",
      "305015 0\n",
      "273815 0\n",
      "279615 0\n",
      "313215 0\n",
      "252915 0\n",
      "234415 0\n",
      "253615 0\n",
      "258115 0\n",
      "233915 3\n",
      "316215 0\n",
      "247715 0\n",
      "260715 0\n",
      "301315 0\n",
      "299215 0\n",
      "242315 0\n",
      "320915 0\n",
      "334615 0\n",
      "233115 0\n",
      "284915 1\n",
      "279115 0\n",
      "331615 0\n",
      "271315 0\n",
      "298315 0\n",
      "237915 0\n",
      "297615 0\n",
      "291415 0\n",
      "271215 0\n",
      "318015 0\n",
      "332115 1\n",
      "320115 0\n",
      "296315 0\n",
      "249715 0\n",
      "306115 0\n",
      "296815 0\n",
      "314115 1\n",
      "333015 0\n",
      "315215 0\n",
      "285415 0\n",
      "253315 0\n",
      "280215 0\n",
      "298515 0\n",
      "329415 0\n",
      "317915 0\n",
      "319815 0\n",
      "254015 0\n",
      "275515 0\n",
      "242515 0\n",
      "258415 0\n",
      "274915 0\n",
      "323615 0\n",
      "241315 0\n",
      "321915 0\n",
      "304215 0\n",
      "321215 0\n",
      "333815 0\n",
      "296515 0\n",
      "278715 0\n",
      "246415 0\n",
      "280115 0\n",
      "291915 0\n",
      "254515 0\n",
      "285615 0\n",
      "251115 0\n",
      "316015 0\n",
      "237015 0\n",
      "292515 0\n",
      "237115 0\n",
      "322115 0\n",
      "254215 1\n",
      "233715 0\n",
      "280615 0\n",
      "290415 3\n",
      "327315 0\n",
      "310415 0\n",
      "282315 0\n",
      "312015 0\n",
      "326315 0\n",
      "290815 0\n",
      "314215 0\n",
      "284115 0\n",
      "236915 0\n",
      "254715 0\n",
      "329315 0\n",
      "238215 0\n",
      "328215 0\n",
      "245015 0\n",
      "289715 0\n",
      "331815 0\n",
      "281315 0\n",
      "252515 0\n",
      "242115 0\n",
      "299615 0\n",
      "281915 0\n",
      "327115 0\n",
      "319915 0\n",
      "249315 1\n",
      "327815 0\n",
      "233815 0\n",
      "233515 0\n",
      "284815 0\n",
      "289515 0\n",
      "313115 0\n",
      "312915 0\n",
      "255115 0\n",
      "317115 0\n",
      "277015 0\n",
      "236515 0\n",
      "233315 3\n",
      "247215 0\n",
      "313015 0\n",
      "247115 1\n",
      "240715 0\n",
      "260415 0\n",
      "293915 0\n",
      "259115 0\n",
      "252615 0\n",
      "324915 0\n",
      "241215 0\n",
      "236115 0\n",
      "241815 0\n",
      "324115 4\n",
      "240215 0\n",
      "328815 1\n",
      "257415 0\n",
      "323115 0\n",
      "249615 1\n",
      "248115 0\n",
      "314715 0\n",
      "254115 0\n",
      "259015 0\n",
      "289215 0\n",
      "247515 0\n",
      "300415 0\n",
      "293815 0\n",
      "301715 0\n",
      "303815 0\n",
      "260915 0\n",
      "257515 0\n",
      "283015 0\n",
      "277515 0\n",
      "272615 0\n",
      "320215 0\n",
      "331515 0\n",
      "309115 0\n",
      "301815 0\n",
      "260315 0\n",
      "260515 0\n",
      "303615 0\n",
      "246315 0\n",
      "288215 0\n",
      "273615 0\n",
      "275915 0\n",
      "250215 0\n",
      "323215 0\n",
      "297015 4\n",
      "259915 0\n",
      "249915 0\n",
      "278015 0\n",
      "304415 0\n",
      "282915 0\n",
      "282215 0\n",
      "280315 0\n",
      "324215 0\n",
      "285915 0\n",
      "272915 0\n",
      "261815 0\n",
      "295715 0\n",
      "236715 0\n",
      "326215 0\n",
      "275715 0\n",
      "333715 0\n",
      "290015 0\n",
      "325715 0\n",
      "333615 0\n",
      "234315 0\n",
      "234815 0\n",
      "286415 0\n",
      "285815 0\n",
      "308415 0\n",
      "284615 0\n",
      "311615 0\n",
      "284715 1\n",
      "296115 0\n",
      "239615 3\n",
      "306415 0\n",
      "308515 0\n",
      "261615 0\n",
      "316115 0\n",
      "274015 0\n",
      "291115 0\n",
      "261015 0\n",
      "294315 0\n",
      "295115 0\n",
      "278215 0\n",
      "300915 0\n",
      "292315 0\n",
      "330215 1\n",
      "258915 0\n",
      "334215 0\n",
      "319415 0\n",
      "325015 0\n",
      "309515 0\n",
      "293715 0\n",
      "279415 0\n",
      "281215 0\n",
      "307415 0\n",
      "324015 0\n",
      "324815 0\n",
      "320615 0\n",
      "322615 0\n",
      "296615 1\n",
      "298815 0\n",
      "316915 0\n",
      "310615 0\n",
      "259615 0\n",
      "307215 0\n",
      "239315 0\n",
      "303015 0\n",
      "327915 0\n",
      "308715 1\n",
      "294015 1\n",
      "273515 0\n",
      "317715 0\n",
      "234115 0\n",
      "276315 0\n",
      "246915 0\n",
      "309015 0\n",
      "234015 0\n",
      "271815 0\n",
      "288915 0\n",
      "305815 0\n",
      "241515 0\n",
      "288715 0\n",
      "319315 0\n",
      "274715 0\n",
      "315715 0\n",
      "256915 0\n",
      "238615 0\n",
      "323515 0\n",
      "295215 0\n",
      "250915 0\n",
      "328515 0\n",
      "318415 0\n",
      "313715 0\n",
      "334715 0\n",
      "233615 0\n",
      "307715 0\n",
      "324415 0\n",
      "327515 0\n",
      "261715 0\n",
      "290215 0\n",
      "314415 0\n",
      "282015 0\n",
      "322815 0\n",
      "328915 1\n",
      "257215 0\n",
      "272815 0\n",
      "252815 0\n",
      "257615 0\n",
      "243315 5\n",
      "274515 0\n",
      "281615 0\n",
      "261115 0\n",
      "254615 0\n",
      "256015 0\n",
      "256715 0\n",
      "333915 0\n",
      "299415 0\n",
      "272215 0\n",
      "326415 0\n",
      "283215 1\n",
      "242815 0\n",
      "328015 0\n",
      "279815 0\n",
      "332515 1\n",
      "315915 0\n",
      "254415 0\n",
      "243415 5\n",
      "274315 0\n",
      "325915 6\n",
      "256615 0\n",
      "288515 0\n",
      "293215 0\n",
      "250815 0\n",
      "296715 0\n",
      "303115 0\n",
      "333115 0\n",
      "239215 0\n",
      "327715 0\n",
      "239815 0\n",
      "316615 0\n",
      "247915 0\n",
      "298415 0\n",
      "250615 0\n",
      "287015 0\n",
      "314015 0\n",
      "246115 0\n",
      "302515 0\n",
      "315515 1\n",
      "243215 4\n",
      "246615 0\n",
      "331215 0\n",
      "315415 1\n",
      "326115 0\n",
      "248015 0\n",
      "238915 0\n",
      "297415 0\n",
      "321015 0\n",
      "287415 0\n",
      "295615 0\n",
      "297215 1\n",
      "302215 0\n",
      "238715 0\n",
      "312315 0\n",
      "275815 0\n",
      "273315 0\n",
      "259815 0\n",
      "256215 0\n",
      "253515 0\n",
      "276415 0\n",
      "287215 0\n",
      "324615 0\n",
      "271615 0\n",
      "241715 1\n",
      "328315 1\n",
      "240015 0\n",
      "238515 0\n",
      "256815 1\n",
      "305115 3\n",
      "320715 0\n",
      "307915 0\n",
      "311415 0\n",
      "297315 0\n",
      "331115 0\n",
      "301915 0\n",
      "332015 0\n",
      "277715 0\n",
      "330715 0\n",
      "252715 0\n",
      "326815 0\n",
      "282115 0\n",
      "239715 0\n",
      "321715 0\n",
      "284415 0\n",
      "294715 0\n",
      "258215 0\n",
      "298715 0\n",
      "259315 1\n",
      "284315 0\n",
      "295415 0\n",
      "252015 0\n",
      "249815 0\n",
      "302715 0\n",
      "275415 0\n",
      "306815 0\n",
      "289115 0\n",
      "316515 3\n",
      "303915 0\n",
      "330015 0\n",
      "287815 1\n",
      "305415 2\n",
      "237715 0\n",
      "282815 0\n",
      "258315 0\n",
      "332615 0\n",
      "331715 0\n",
      "298015 0\n",
      "309415 0\n",
      "308115 0\n",
      "289315 0\n",
      "271115 0\n",
      "314315 0\n",
      "274215 0\n",
      "280915 0\n",
      "246215 0\n",
      "255815 0\n",
      "242915 0\n",
      "253115 0\n",
      "252115 0\n",
      "327615 0\n",
      "249415 0\n",
      "240915 0\n",
      "236015 0\n",
      "300715 0\n",
      "239115 0\n",
      "298915 0\n",
      "293515 0\n",
      "278515 0\n",
      "299515 0\n",
      "258515 0\n",
      "272115 0\n",
      "328415 0\n",
      "301215 0\n",
      "277315 0\n",
      "292015 0\n",
      "288615 0\n",
      "237315 0\n",
      "311915 0\n",
      "259415 0\n",
      "276015 0\n",
      "312115 0\n",
      "285115 0\n",
      "320815 0\n",
      "330815 0\n",
      "276515 0\n",
      "311215 0\n",
      "288815 0\n",
      "235215 0\n",
      "328115 0\n",
      "309715 0\n",
      "318315 0\n",
      "275615 0\n",
      "306015 1\n",
      "249115 0\n",
      "292715 0\n",
      "295815 0\n",
      "273115 0\n",
      "290515 0\n",
      "288015 0\n",
      "316715 0\n",
      "255415 0\n",
      "310215 0\n",
      "271415 1\n",
      "248515 1\n",
      "240515 0\n",
      "330915 0\n",
      "259215 0\n",
      "254315 0\n",
      "251315 0\n",
      "246015 0\n",
      "257715 0\n",
      "289815 0\n",
      "328615 0\n",
      "235415 0\n",
      "310315 0\n",
      "297815 0\n",
      "315015 0\n",
      "250115 0\n",
      "297715 0\n",
      "308015 0\n",
      "255015 0\n",
      "316415 0\n",
      "311015 0\n",
      "315815 0\n",
      "253015 0\n",
      "315115 7\n",
      "325215 0\n",
      "314515 0\n",
      "321515 0\n",
      "277415 0\n",
      "238815 0\n",
      "232915 0\n",
      "281715 0\n",
      "279715 0\n",
      "292615 0\n",
      "319715 0\n",
      "297515 0\n",
      "299315 0\n",
      "306315 0\n",
      "292415 0\n",
      "294815 0\n",
      "318515 0\n",
      "286915 0\n",
      "251815 0\n",
      "253215 0\n",
      "253415 0\n",
      "332215 1\n",
      "276715 0\n",
      "329215 0\n",
      "283415 0\n",
      "319615 0\n",
      "322515 0\n",
      "320415 0\n",
      "240615 0\n",
      "286515 0\n",
      "321815 0\n",
      "304915 0\n",
      "247015 0\n",
      "326615 0\n",
      "293315 0\n",
      "301615 0\n",
      "313515 0\n",
      "329115 0\n",
      "236415 0\n",
      "307615 1\n",
      "281515 0\n",
      "307315 1\n",
      "318115 0\n",
      "281415 1\n",
      "321415 0\n",
      "279915 0\n",
      "330315 0\n",
      "274415 0\n",
      "307815 0\n",
      "321315 0\n",
      "311315 0\n",
      "235815 0\n",
      "322915 0\n",
      "274615 0\n",
      "282415 0\n",
      "234515 0\n",
      "281115 0\n",
      "248715 0\n",
      "250715 0\n",
      "271915 0\n",
      "237415 0\n",
      "303415 0\n",
      "236215 0\n",
      "297915 0\n",
      "274115 0\n"
     ]
    }
   ],
   "source": [
    "for k,i in shortjobs.items():\n",
    "    print(k,len(i))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bundle_samples(jobdict, minsize=50, simplename=False):\n",
    "    bundledict = {} #bundled jobdict\n",
    "    bundle_name = \"\"\n",
    "    current_jobs = []\n",
    "    bundlecount = 1\n",
    "    for sample, jobs in jobdict.items():\n",
    "        if len(jobs)>0:\n",
    "            #print(sample)\n",
    "            current_jobs = current_jobs+jobs\n",
    "            if simplename == True:\n",
    "                bundle_name = 'bundle{}'.format(str(bundlecount))\n",
    "            else:\n",
    "                bundle_name = bundle_name+\"_\"+sample\n",
    "            if len(current_jobs)>minsize:\n",
    "                bundledict[bundle_name] = current_jobs\n",
    "                bundle_name=\"\"\n",
    "                current_jobs=[]\n",
    "                bundlecount = bundlecount+1\n",
    "    bundledict[bundle_name] = current_jobs\n",
    "    return bundledict\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "#bdict['_290315_297115_282615_298215_321615_302815_327215_286815_310915_311115_305915_288115_246715_333515_308215_304015_325415_301515_302315_313315_305515_271515_238415_260215_309215_251015_258015_256415_244815_291615_248915_318715_325515_235315_257015_296015_295315_235015_283515_303715_315615_237815_283615_236315_307115_302415_320015_322315_271715_285515_291715']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "bdict = bundle_samples(jobdict=shortjobs, minsize=150, simplename=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "bdict2= bundle_samples(jobdict=jobdict, minsize=150)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " 0\n"
     ]
    }
   ],
   "source": [
    "for key, item in bdict2.items():\n",
    "    print( key, len(item))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "bundle1 131\n"
     ]
    }
   ],
   "source": [
    "for key, item in bdict.items():\n",
    "    print( key, len(item))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "_244915_245515_243815_244515_244215_243715\n",
      "172\n",
      "_244715_244415_245415_245215_244315_244615\n",
      "170\n",
      "_245815_247815_245315\n",
      "65\n",
      "bundle1\n",
      "131\n"
     ]
    }
   ],
   "source": [
    "create_batchfiles(jobdict=bdict2, scriptdir=\"/home/tilman/storage2/process_stripes/stripes_optimise_F15/Stripes/stripes\", prefix=\"bundles_200421_200_002_\")\n",
    "create_batchfiles(jobdict=bdict, scriptdir=\"/home/tilman/storage2/process_stripes/stripes_optimise_F15/Stripes/stripes\", prefix=\"bundles_200421_200_003_\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "#present_all"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
